### ISSUE TITLE
Resizing window breaks layout

### ISSUE THREAD

**Issue by teawaterwire** [reapp/reapp#23]
Hi!

Just be playing with the base reapp stack, and noticed that when manually resizing my browser's window the layout would break on subsequent click on "Go to sub view"

Cheers

**Comment by natew** [comment]
This is true, and not a hard fix. Just haven't got around to it yet as mobile devices don't do resizes and that's our initial target. Will keep this open as we want to fix it eventually.

**Comment by teawaterwire** [comment]
Don't mobile devices do resizes when switching to landscape mode?
But I agree this is not critical :)

**Comment by natew** [comment]
Ah yes, agreed. If you want to look at fixing it, all you'd need to check is ViewListMixin and the scroller setDimensions. But yes, hopefully in the next month we can tackle this.

**Comment by clauderic** [comment]
Any progress on this? If not, could you expose a little more details on how it could work if you had a listener on orientation change or on resize? I do agree with @teawaterwire that this is necessary on device orientation change

**Comment by jvanderberg** [comment]
This is actually a pretty critical issue for us as orientation change is a pretty mainstream use case on mobile.  I've looked at the ViewListMixin code and it's not clear to me where the simple fix lies - can you provide a bit more direction?

**Comment by clauderic** [comment]
I spent a pretty long time trying to get it to work on my end. You have to modify [ViewListMixin](https://github.com/reapp/reapp-ui/blob/master/src/views/ViewListMixin.jsx), specifically, the resize method, you need to do something like this on resize, as a callback to the setState update of this.setupDimensions():

```javascript
this.stepper.width = this.state.width;
this.scroller.setDimensions(this.state.width, null, this.state.width * that.props.children.length, null, false);
```

It works, but in the end I decided to write my own NestedViewList / View / TitleBar components using percentage values for widths and translate3d offsets and CSS transitions for animations. I did this for a couple of reasons. For starters, CSS transitions/animations are much more performant on iOS than relying on javascript for animations, especially if you're using WKWebView, and it's responsive from the get-go since it relies on css % values instead of cached px values in the React state. It also gets around the problem that reapp prevents setState to be called on the children of NestedViewList Views until all animations have completed, because it relies on setState for animation purposes (see https://github.com/reapp/reapp-ui/issues/68)
